### PR TITLE
[SPARK-14548][SQL] Support not greater than and not less than operator in Spark SQL

### PR DESCRIPTION
**Pull request by jliwork**
## What changes were proposed in this pull request?

!< means not less than which is equivalent to >=
!> means not greater than which is equivalent to <=

I'd to create a PR to support these two operators. 
## How was this patch tested?

I've added new test cases in: DataFrameSuite, ExpressionParserSuite, JDBCSuite, PlanParserSuite, SQLQuerySuite

@dilipbiswal @viirya @gatorsmile

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala [hunk 1]
```diff
@@ -1432,4 +1432,15 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       getMessage()
     assert(e1.startsWith("Path does not exist"))
   }
+
+  test("test !< and !> comparator") {
```
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala]: you probably don't need this test case

**Comment by viirya** [comment]
@rxin Do we need to add these operators to dsl?

**Comment by rxin** [comment]
No let's not do that.

**Comment by viirya** [comment]
Can you trigger the jenkin test? Thanks.

**Comment by gatorsmile** [comment]
FYI, MS SQL Server supports these operators: https://msdn.microsoft.com/en-us/library/ms188074.aspx

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala (hunk 1 above)
Comment (jliwork) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala]: Hi @rxin, 
Thanks for your comment! I've removed this test case from DataFrameSuite. Please free free to let me know of any other comments or questions.

**Comment by hvanhovell** [comment]
I am not nessecarily against this but I find these quite confusing. Why not just write `>=` or `<=`?

The PR itself looks pretty solid.

**Comment by rxin** [comment]
Can you summarize the list of database systems that support this syntax?

**Comment by jliwork** [comment]
@rxin Microsoft SQL Server, IBM DB2 and Sybase Adaptive Server support them. 

https://msdn.microsoft.com/en-us/library/ms188074.aspx
http://www.ibm.com/support/knowledgecenter/SSEPGG_10.5.0/com.ibm.db2.luw.sql.ref.doc/doc/r0000746.html?lang=en
http://infocenter.sybase.com/archive/index.jsp?topic=/com.sybase.dc36271_36272_36273_36274_1250/html/refman/refman237.htm

**Comment by jliwork** [comment]
@rxin gentle ping

**Comment by rxin** [comment]
@hvanhovell Can you think of any downsides if we support this? It is a little bit esoteric. The main one I can think of is whether this would set a precedent in that a lot of people start to submit "compatibility" things for esoteric features for different databases.

**Comment by hvanhovell** [comment]
@rxin it being a bit on the esoteric side and setting a precedence are my minor objections. I am fine putting this in given the size and impact this has on the code base.

**Comment by rxin** [comment]
OK thanks - I'm going to merge this. The test coverage is too excessive here. I'm going to remove most of the test cases before merging.

**Comment by jliwork** [comment]
Awesome. Thank you! :-)
